### PR TITLE
Feature/container uid

### PR DESCRIPTION
**Pull request by mblaschke**
- Add support for eg. `docker run --user 999 webdevops/php-apache:ubuntu-16.04`

### CODE REVIEW

**Comment by hhoechtl** [comment]
I'm not quite sure if that's within the concept of docker, because I don't know of any other image having that functionality. Usually you have a fixed user running everything inside the container e.g. nginx, elasticsearch, postgres, etc. 
But implementation looks good though.

**Comment by mblaschke** [comment]
The master process of FPM needs AFAIK root rights for eg. slowlog so it should be started as root.
Also most of the configuration files are not changeable for other users for security reasons: An attacker (via FPM) can modify configuration files or other files.

So i'm detecting that docker container is not running as root, privilege escalation via `gosu` and remove the suid bit after privilege escalation. So the main entrypoint is still running via root but the application can be run as normal user.
If we don't use privilege escalation then all processes must be able to run under this user which will be more difficult.

With this fix we can run FPM under any user and new files are also created under this UID and is not fixed to UID 1000 anymore.

**Comment by mblaschke** [comment]
eg. FPM configuration and all other configuration which are modified after ENTRYPOINT startup (eg. by env vars) must be world writeable if we don't do privilege escalation. I'm really not happy with world writeable configuration files.

File: docker/bootstrap/ubuntu-14.04/Dockerfile [hunk 1]
```diff
@@ -48,6 +48,19 @@ RUN export DEBIAN_FRONTEND=noninteractive \
...
+    && dpkgArch="amd64" \
+    && wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch" \
+    && wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc" \
+    && export GNUPGHOME="$(mktemp -d)" \
+    && gpg --keyserver ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 \
+    && gpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu \
+    && rm -r "$GNUPGHOME" /usr/local/bin/gosu.asc \
+    && chmod +x /usr/local/bin/gosu \
+    && gosu nobody true \
+    && apt-get purge -y -f --force-yes wget \
```
Comment (sergeyklay) [inline docker/bootstrap/ubuntu-14.04/Dockerfile]: How about adding `wget` in next section (below)

File: docker/bootstrap/ubuntu-14.04/Dockerfile [hunk 2]
```diff
@@ -48,6 +48,19 @@ RUN export DEBIAN_FRONTEND=noninteractive \
     && hash -r \
     && pip install ansible \
     && chmod 750 /usr/local/bin/ansible* \
+    # Install gosu
+    && GOSU_VERSION=1.10 \
+    && /usr/local/bin/apt-install wget gnupg \
+    && dpkgArch="amd64" \
+    && wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch" \
+    && wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc" \
```
Comment (sergeyklay) [inline docker/bootstrap/ubuntu-14.04/Dockerfile]:
```
wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$(dpkg --print-architecture | awk -F- '{ print $NF }').asc"
```

Comment (mblaschke) [inline docker/bootstrap/ubuntu-14.04/Dockerfile]: I've only used the part from the "official" installation and wanted to keep the steps :)
https://github.com/tianon/gosu

Comment (sergeyklay) [inline docker/bootstrap/ubuntu-14.04/Dockerfile]: https://github.com/tianon/gosu#from-debian

**Comment by hhoechtl** [comment]
@mblaschke ok, I'll get it. Brilliant solution!

**Pull request merged**